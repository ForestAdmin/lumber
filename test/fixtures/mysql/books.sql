CREATE TABLE books (
  id INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
  title VARCHAR(100) NOT NULL,
  author VARCHAR(100) NOT NULL,
  published_date DATE NOT NULL,
  isbn INT,
  CONSTRAINT books_isbn_key UNIQUE (isbn ASC)
);
