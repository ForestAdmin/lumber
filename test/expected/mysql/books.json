{
  "fields":[
    {
      "name":"title",
      "nameColumn":"title",
      "type":"STRING",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"author",
      "nameColumn":"author",
      "type":"STRING",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"publishedDate",
      "nameColumn":"published_date",
      "type":"DATE",
      "primaryKey":false,
      "defaultValue":null
    },
    {
      "name":"isbn",
      "nameColumn":"isbn",
      "type":"INTEGER",
      "primaryKey":false,
      "defaultValue":null
    }
  ],
  "references":[
    {
      "isHasOneOrHasMany":true,
      "ref":"reviews",
      "association":"hasMany"
    },
    {
      "isBelongsToMany":true,
      "ref":"users_books",
      "association":"belongsToMany",
      "junctionTable":"users"
    }
  ],
  "primaryKeys":["id"],
  "options":{
    "underscored":true,
    "timestamps":false,
    "hasIdColumn":true,
    "hasPrimaryKeys":true
  }
}